### ISSUE TITLE
Typescript error due to import declaration conflict (breaking change in Typescript v3.7)

### ISSUE THREAD

**Issue by vicrep** [redux-saga/redux-saga#1971]
### Steps to reproduce
- run TSC using typescript ≥ 3.7 on a project using redux-saga (without `skipLibCheck` enabled)
- output:
```
node_modules/@redux-saga/core/types/ts3.6/index.d.ts:2:33 - error TS2440: Import declaration conflicts with local declaration of 'END'.

2 import { Saga, Buffer, Channel, END, Predicate, SagaIterator, Task } from '@redux-saga/types'
                                  ~~~
```

#### Description of the bug/issue

This is directly caused by https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#local-and-imported-type-declarations-now-conflict

Would open a PR to fix, but don't appreciate scope of declarations enough to do it sensibly.

### Environment information

- redux-saga version: 1.1.2
- babel / typescript setup: typescript: 3.7.2

**Comment by Andarist** [comment]
Thanks for the report, I was sure that 1.1.2 has resolved this - but apparently the fix was only applied to typings targeting older versions of TS. I've made the same fix for >=3.6 typings right now - hopefully this should resolve the issue entirely. Please install 1.1.3.

**Comment by pedrosimao** [comment]
I have installed 1.1.3 and I still have the same issue...
